### PR TITLE
Updated Quick, Nimble and ReactiveObjc for swift 4.2 support

### PR DESCRIPTION
**Pull request by jakovi89**
* Currently pointing at an intermediate branch  for ReactiveObjc, supporting Swift 4.2
* Updates Quick and Nimble to Swift 4.2 versions
* Updates CI to Xcode 10.2

### CODE REVIEW

**Comment by mdiep** [comment]
CI failed. It looks like this PR still needs some work.

**Comment by jakovi89** [comment]
hmm.. Watch OS still fails, I'm looking into it

**Comment by jakovi89** [comment]
I'm somewhat in doubt why the Watch target will try to compile the UITextView+Rac on the CI, since these isn't part of the "compile sources". @mdiep do you have an idea why?

**Comment by jakovi89** [comment]
It seems that the WatchOS target isn't choosing the correct target from ReactiveObjc

**Comment by mdiep** [comment]
This is probably the old Xcode bug where it doesn't pick the correct platform for some reason. In the past, we've gotten around this by turning off implicit dependencies in the scheme and making the dependencies explicit.

**Comment by jakovi89** [comment]
Thanks for the hint @mdiep 👍

File: Cartfile [hunk 1]
```diff
@@ -1,2 +1,2 @@
-github "ReactiveCocoa/ReactiveObjC" ~> 3.1.0
-github "ReactiveCocoa/ReactiveSwift" ~> 4.0.0
+github "ReactiveCocoa/ReactiveObjC" "27a7254837bb01122b51aac7c4c3961973351ca0"
```
Comment (mdiep) [inline Cartfile]: Actually, we should get an actual release here

Comment (mdiep) [reply Cartfile]: I just released 3.1.1. Please update the Cartfile with that.
https://github.com/ReactiveCocoa/ReactiveObjC/releases/tag/3.1.1

**Comment by mdiep** [comment]
Thanks! I made a new release with this: https://github.com/ReactiveCocoa/ReactiveObjCBridge/releases/tag/5.0.0

**Pull request merged**